sdks/python/apache_beam/io/avroio.py

```diff
@@ -66,19 +68,22 @@ def __init__(self, label=None, file_pattern=None, min_bundle_size=0,
     Args:
       label: label of the PTransform.
       file_pattern: the set of files to be read.
+      coder: a coder to use to decode values (iff the record type is BYTES).
```

Comment:
I agree to the general point you mentioned here, that coders may be used to generate elements produced by sources (note that some sources such as Java XmlSource, AvroSource don't do this).

My point was that, if the schema of Avro file is ''{"type": "bytes"}', that source should be producing bytes. Using a coder (or a function) to convert that to a different type is a functionality specific to AvroSource, that we are trying to add, which I believe is error prone. 

I'm OK with adding this with clear documentation, and runtime error checking, if there are valid use cases.